oak_functions/location_utils/src/tests.rs

```diff
@@ -16,165 +16,114 @@
+    let location = location_from_degrees(51.5074_f64, -0.1278_f64);
+    // Level 2.
+    let level = 2;
+    let cell = find_cell(&location, level).unwrap();
+    let expected = CellID::from(&location).parent(level as u64);
 
-    // Check relative to the London cell.
-    let cell = find_cell(london_lat, london_lon).unwrap();
-    let london = cell.relative_position(london_lat, london_lon);
-    let oxford = cell.relative_position(oxford_lat, oxford_lon);
+    assert_eq!(cell, expected);
 
-    let distance = (london.squared_distance(&oxford) as f32).sqrt() as i32;
+    // Level 7.
+    let level = 7;
+    let cell = find_cell(&location, level).unwrap();
+    let expected = CellID::from(&location).parent(level as u64);
 
-    // Allow up to 10m distortion from the projection at this distance.
-    assert!((distance - hav_distance).abs() <= 10);
+    assert_eq!(cell, expected);
 
-    // Check relative to the Oxford cell.
-    let cell = find_cell(oxford_lat, oxford_lon).unwrap();
-    let london = cell.relative_position(london_lat, london_lon);
-    let oxford = cell.relative_position(oxford_lat, oxford_lon);
+    // Level 20.
+    let level = 20;
+    let cell = find_cell(&location, level).unwrap();
+    let expected = CellID::from(&location).parent(level as u64);
 
-    let distance = (london.squared_distance(&oxford) as f32).sqrt() as i32;
+    assert_eq!(cell, expected);
 
-    // Allow up to 10m distortion from the projection at this distance.
-    assert!((distance - hav_distance).abs() <= 10);
-}
+    // Level 30.
+    let level = 30;
+    let cell = find_cell(&location, level).unwrap();
+    let expected = CellID::from(&location);
 
-/// Calculates the distance between two points on a spherical approximation of the earth using the
-/// Haversine formula.
-///
-/// See https://en.wikipedia.org/wiki/Haversine_formula
-fn haversine_distance(lat1: f32, lon1: f32, lat2: f32, lon2: f32) -> f32 {
-    let lat1_radians = lat1.to_radians();
-    let lat2_radians = lat2.to_radians();
-    let h = haversine(lat2_radians - lat1_radians)
-        + lat1_radians.cos()
-            * lat2_radians.cos()
-            * haversine(lon2.to_radians() - lon1.to_radians());
-    2.0 * EARTH_RADIUS * h.sqrt().atan()
+    assert_eq!(cell.level(), level as u64);
+    assert_eq!(cell, expected);
 }
 
-fn haversine(theta: f32) -> f32 {
-    (theta / 2.0).sin() * (theta / 2.0).sin()
+#[test]
+fn test_find_covering_cells() {
+    let location = location_from_degrees(51.5074_f64, -0.1278_f64);
+
+    // We should only need 2 level 5 cell.
+    // See https://s2.sidewalklabs.com/regioncoverer/?center=51.504569%2C-0.128076&zoom=8&cells=47dc,4874
+    // for a visualisation.
+    let cover =
+        find_covering_cells(&location, &Angle::from(DEFAULT_CUTOFF_RADIUS_RADIANS), 5).unwrap();
+
+    assert_eq!(cover.len(), 2);
+    assert_eq!(cover[0].to_token(), "47dc");
+    assert_eq!(cover[1].to_token(), "4874");
+
+    // We should need 4 level 6 cells.
```

Comment:
```suggestion
    // We should need 4 level-6 cells.
```
Should it be hyphenated?